File: scripts/parse_presentation.py
from bs4 import BeautifulSoup

def parsePresentation(unpack_path):
    presentation_path = unpack_path+"/ppt/presentation.xml"
    soup = BeautifulSoup(open(presentation_path, 'r').read(), 'xml')
    slide_size_tag = soup.find('sldSz')
    return slide_size_tag['cx'], slide_size_tag['cy']

File: src/ParsePresentation.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.parser.Parser;

public class ParsePresentation {
	public String[] retrieveDimensions(String unpack_path) {
		try {
			byte[] encoded = Files.readAllBytes(Paths.get(unpack_path + "/ppt/presentation.xml"));
			
			Document soup = Jsoup.parse(new String(encoded), "", Parser.xmlParser());
			Element slide_size_tag = soup.select("p|sldSz").get(0);
			
			return new String[] {
					slide_size_tag.attr("cx"),
					slide_size_tag.attr("cy")
			};
		} catch (IOException e) {
			System.out.println("Corrupted xml file " + unpack_path);
			System.exit(1);
		}
		// It will never be reached.
		return null;
	}
}
